File: airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/credential/GcsCredentialConfigs.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.gcs.credential;

import com.fasterxml.jackson.databind.JsonNode;
import io.airbyte.commons.json.Jsons;

public class GcsCredentialConfigs {

  public static GcsCredentialConfig getCredentialConfig(final JsonNode config) {
    final JsonNode credentialConfig = config.get("credential");
    final GcsCredentialType credentialType = GcsCredentialType.valueOf(credentialConfig.get("credential_type").asText().toUpperCase());

    if (credentialType == GcsCredentialType.HMAC_KEY) {
      return new GcsHmacKeyCredentialConfig(credentialConfig);
    }
    throw new RuntimeException("Unexpected credential: " + Jsons.serialize(credentialConfig));
  }

}

File: airbyte-cdk/java/airbyte-cdk/gcs-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/gcs/credential/GcsCredentialConfigs.kt
/*
 * Copyright (c) 2026 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.destination.gcs.credential

import com.fasterxml.jackson.databind.JsonNode
import java.util.*

object GcsCredentialConfigs {
    fun getCredentialConfig(config: JsonNode): GcsCredentialConfig {
        val credentialConfig = config["credential"]
        val credentialType =
            GcsCredentialType.valueOf(
                credentialConfig["credential_type"].asText().uppercase(Locale.getDefault())
            )

        return when (credentialType) {
            GcsCredentialType.HMAC_KEY -> GcsHmacKeyCredentialConfig(credentialConfig)
            GcsCredentialType.SERVICE_ACCOUNT -> GcsServiceAccountCredentialConfig(credentialConfig)
        }
    }
}
